### PR TITLE
Rename "NewRelic.php" to "Config.php"

### PR DESCRIPTION
**Pull request by Nyholm**
This is something that has bothered me for long. Pretty much every time I look at the source Im like "What does the NewRelic class do?"

I've also added a few `declare(strict_types=1);` and changed privacy from "protected" to "private".

### CODE REVIEW

File: DependencyInjection/EkinoNewRelicExtension.php [hunk 1]
```diff
@@ -59,7 +60,7 @@ public function load(array $configs, ContainerBuilder $container)
             $config['deployment_names'] = array_values(array_filter(explode(';', $config['application_name'])));
         }
 
-        $container->getDefinition('ekino.new_relic')
+        $container->getDefinition(Config::class)
```
Comment (Nyholm) [inline DependencyInjection/EkinoNewRelicExtension.php]: Should we make this change to more services?

Comment (jderusse) [reply DependencyInjection/EkinoNewRelicExtension.php]: This is the new way to declare services. I'm :+1:

**Comment by Nyholm** [comment]
I also dropped `getFramework`... It was never used.

File: Listener/RequestListener.php [hunk 2]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare(strict_types=1);
```
Comment (jderusse) [inline Listener/RequestListener.php]: Could be defined in .php_cs file ? (Same for private methods)

Comment (Nyholm) [reply Listener/RequestListener.php]: What do you mean? Do Travis will fail if we do not have this line?

File: DependencyInjection/EkinoNewRelicExtension.php (hunk 1 above)
Comment (Nyholm) [reply DependencyInjection/EkinoNewRelicExtension.php]: I added: https://github.com/ekino/EkinoNewRelicBundle/issues/160

File: Listener/RequestListener.php (hunk 2 above)
Comment (jderusse) [reply Listener/RequestListener.php]: Yes m'y point is to switch all files to strict. And add the rule in php_cs to not forget it.

File: Listener/RequestListener.php [hunk 3]
```diff
@@ -20,33 +22,54 @@
...
 
-    protected $interactor;
+    /**
+     * @var NewRelicInteractorInterface
+     */
+    private $interactor;
 
-    protected $transactionNamingStrategy;
+    /**
+     * @var TransactionNamingStrategyInterface
```
Comment (jderusse) [inline Listener/RequestListener.php]: IDE are able to guess type from the constructor. Do we have to document all those parameters? If yes, what's about the class Config?

File: Listener/RequestListener.php [hunk 4]
```diff
@@ -20,33 +22,54 @@
 
 class RequestListener
 {
-    protected $ignoredRoutes;
+    /**
+     * @var array
```
Comment (jderusse) [inline Listener/RequestListener.php]: `string[[` brings more value que `array` IMHO

File: Listener/RequestListener.php (hunk 3 above)
Comment (Nyholm) [reply Listener/RequestListener.php]: You are correct.

File: Listener/RequestListener.php (hunk 2 above)
Comment (Nyholm) [reply Listener/RequestListener.php]: 👍  Please send a PR with that

**Comment by Nyholm** [comment]
I've updated the PR

**Pull request merged**